PRIME-1551 Integration testing with PLR

neophyte57 opened the pull request:
Completed initial end-to-end integration testing with Gary (PLR), with mutual authentication occurring.  

White-listing PLR IP addresses not to be implemented

Can be tested locally like this:  `curl --insecure --header "@../Utils/Auth/client-pem-url-encoded-headers.txt" --data-binary "@PRPM_IN303030CA.xml" https://127.0.0.1:5001/api/PLRHL7`  
![image](https://user-images.githubusercontent.com/7586665/118054724-05b79f00-b344-11eb-811c-4692d8e47673.png)

or remotely like this `curl --insecure --cert client.crt:changeme --key client.key --data-binary "@PRPM_IN303030CA.xml" https://mutualauth-pr-1247.pharmanetenrolment.gov.bc.ca/api/v1/PLRHL7`

johnnyl6 commented on prime-dotnet-webapi/Services/SoapService.cs:
```diff
@@ -151,9 +166,20 @@ private PlrProvider ReadDistributionMessage(string messageContent)
+            if (telephoneNumData != null)
+            {
+                string[] telephoneNumberParts = SplitHL7v3TelecomNumber(RemoveHL7v3TelecomType(telephoneNumData));
+                if (telephoneNumberParts.Length == 2)
```
Is it supposed to be `faxNumberParts.Length == 2` ?

johnnyl6 commented on prime-dotnet-webapi/Services/SoapService.cs:
```diff
@@ -134,12 +134,27 @@ private PlrProvider ReadDistributionMessage(string messageContent)
+            if (faxNumberData != null)
+            {
+                string[] faxNumberParts = SplitHL7v3TelecomNumber(RemoveHL7v3TelecomType(faxNumberData));
+                if (faxNumberData.Length == 2)
+                {
```
Is it supposed to be `faxNumberParts.Length == 2`?

johnnyl6 commented on prime-dotnet-webapi/Services/SoapService.cs:
```diff
@@ -177,7 +203,8 @@ public static string[] SplitHL7v3TelecomNumber(string telecomNumber)
             {
-                throw new ArgumentOutOfRangeException(nameof(telecomNumber), telecomNumber, "Not a 10 digit string");
+                return new string[] { telecomNumber };
+                // throw new ArgumentOutOfRangeException(nameof(telecomNumber), telecomNumber, "Not a 10 digit string");
```
Seems safe to remove the comment

neophyte57 commented:
Good catch @johnnyl6 .  I addressed your comments and also a warning in the back-end log "The property 'Credentials' on entity type 'PlrProvider' is a collection or enumeration type with a value converter but with no value comparer. Set a value comparer to ensure the collection/enumeration elements are compared correctly."

johnnyl6 reviewed (approved):
LGTM

mtpultz reviewed (approved):
![image](https://user-images.githubusercontent.com/599498/118169715-1d436600-b3de-11eb-9b3e-7fc960bca789.png)
